File: Python/257. Binary Tree Paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        ans = list()
        if root is None:
            return []
        def dfs(node, s):
            if node.left is None and node.right is None:
                ans.append(s+str(node.val))
            else:
                if node.left:
                    dfs(node.left,s+str(node.val)+'->')
                if node.right:
                    dfs(node.right, s+str(node.val)+'->')
        dfs(root,'')
        return ans

File: C++/257. Binary Tree Paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<string> ans;
    void dfs(TreeNode* node,string s)
    {
        if(node->left==nullptr && node->right==nullptr)
        {
            ans.push_back(s+to_string(node->val));
            return ;
        }
        if(node->left != nullptr)
            dfs(node->left, s+to_string(node->val)+"->");
        if(node->right != nullptr)
            dfs(node->right, s+to_string(node->val)+"->");
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        ans.clear();
        if(root == nullptr) return ans;
        dfs(root,"");
        return ans;
    }
};